archivers/lzlib/Portfile

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                lzlib
+version             1.14
+revision            0
+
+homepage            https://www.nongnu.org/lzip/lzlib.html
+
+categories          archivers devel
+installs_libs       yes
+license             BSD
+maintainers         {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
+
+description         \
+    Lzlib is a data compression library providing in-memory LZMA compression \
+    and decompression functions
+
+long_description        \
+    {*}${description}, including integrity checking of the decompressed data. \
+    The compressed data format used by the library is the lzip format. Lzlib \
+    is written in C. The lzip file format is designed for data sharing and \
+    long-term archiving, taking into account both data integrity and decoder \
+    availability\: The lzip format provides very safe integrity checking and \
+    some data recovery means. The program lziprecover can repair bit flip \
+    errors \(one of the most common forms of data corruption\) in lzip files, \
+    and provides data recovery capabilities, including error-checked merging \
+    of damaged copies of a file. The lzip format is as simple as possible \
+    \(but not simpler\). The lzip manual provides the source code of a simple \
+    decompressor along with a detailed explanation of how it works, so that \
+    with the only help of the lzip manual it would be possible for a digital \
+    archaeologist to extract the data from a lzip file long after quantum \
+    computers eventually render LZMA obsolete. Additionally the lzip \
+    reference implementation is copylefted, which guarantees that it will \
+    remain free forever.
+
+checksums           rmd160  aab868b92c87268ace133d0db2789cb5ab8de8e0 \
+                    sha256  5acac8714ed4f306020bae660dddce706e5f8a795863679037da9fe6bf4dcf6f \
+                    size    131624
+
+master_sites        nongnu:lzip/lzlib
+
+platform darwin {
+    patchfiles-append patch-darwin-Makefile.in.diff
+
+    post-build {
+        set libname liblz.${version}.dylib
+        set libpath ${prefix}/lib/${libname}
+        system "install_name_tool -id ${libpath} ${worksrcpath}/${libname}"
+    }
+}
+
+depends_build-append \
+                    port:help2man
+
+configure.args-append \
+                    --enable-shared \
+                    CC=${configure.cc} \
+                    CFLAGS=${configure.cflags} \
+                    LDFLAGS=${configure.ldflags}
```

Comment:
It really needs these passed as configure arguments? I would take a hard look at whether it can accept these as environment variables instead since MacPorts already sets those.

If you can't make that work, then you must also pass `-arch` flags here.

Reply:
Unfortunately this ignores environment variables.  It's a very old project:

```
➜ sudo -u macports ./configure --help
Usage: ./configure [OPTION]... [VAR=VALUE]...

To assign makefile variables (e.g., CC, CFLAGS...), specify them as
arguments to configure in the form VAR=VALUE.

Options and variables: [defaults in brackets]
  -h, --help            display this help and exit
  -V, --version         output version information and exit
  --srcdir=DIR          find the source code in DIR [. or ..]
  --prefix=DIR          install into DIR [/usr/local]
  --exec-prefix=DIR     base directory for arch-dependent files [$(prefix)]
  --bindir=DIR          user executables directory [$(exec_prefix)/bin]
  --datarootdir=DIR     base directory for doc and data [$(prefix)/share]
  --includedir=DIR      C header files [$(prefix)/include]
  --infodir=DIR         info files directory [$(datarootdir)/info]
  --libdir=DIR          object code libraries [$(exec_prefix)/lib]
  --mandir=DIR          man pages directory [$(datarootdir)/man]
  --disable-static      don't build a static library [enable]
                        (implies --enable-shared)
  --enable-shared       build also a shared library [disable]
  --disable-ldconfig    don't run ldconfig after install
  CC=COMPILER           C compiler to use [gcc]
  AR=ARCHIVER           library archiver to use [ar]
  CPPFLAGS=OPTIONS      command-line options for the preprocessor []
  CFLAGS=OPTIONS        command-line options for the C compiler [-Wall -W -O2]
  CFLAGS+=OPTIONS       append options to the current value of CFLAGS
  LDFLAGS=OPTIONS       command-line options for the linker []
  ARFLAGS=OPTIONS       command-line options for the library archiver [-rcs]
  MAKEINFO=NAME         makeinfo program to use [makeinfo]

```

Those are all the options available.

Reply:
`-arch` flags have been added